### DISCUSSION TITLE
Csla and "Hardware" programming

### DISCUSSION THREAD

**Discussion by Chicagoan2016** [MarimerLLC/csla#1813] [General]
We are looking into developing an application using C# (and hopefully Csla) for a healthcare facility. Could we use C# (and Csla) to talk to a device? I thought .NET compact framework has been inactive since 3.5.
Any ideas please?
Kind Regards

**Comment by rockfordlhotka** [comment]
"talk to a device" could mean many things 😄 

I worked in bio-medical manufacturing for many years, and we had numerous examples of PCs talking to devices - mostly through serial cables (yeah, this was a long time ago) and some through USB.

I've also seen projects where software is _running on a device_, which is a whole other thing. Not just that the tech is different, but as soon as you cross that bridge then a whole host of regulatory issues come into play, because then you are building software that is part of a medical device!!

If we're talking about the first case, where a PC (or other device) (call it C) is communicating with the medical device (call it M) via USB, then the thing to determine is whether .NET can run on C.

.NET today can run on Windows, Linux, and numerous other types of device (Raspberry Pi, Android, iOS, etc.), so it seems likely that the answer is yes - but that needs to be confirmed. And if .NET can run on the device then _probably_ you can write software to communicate via USB.

**Comment by rockfordlhotka** [comment]
@SachinPNikam might have some insight. He recently tweeted about running #cslanet on a small device interacting with hardware.

https://twitter.com/sac_nik/status/1299733281630375939?s=20

**Comment by Chicagoan2016** [comment]
@rockfordlhotka , Thank you Sir.
I found a little more detail, the computer and the medical "devices" (one of them DNA synthesizer) would be connected to the same  LAN. I am assuming the "devices" will have some API I could call from my C# (and hence Csla) code?

Hopefully I won't have to use the nanoframework, has one of fellow Csla developers tried Csla on Nanoframework?

Kind Regards

**Reply by SachinPNikam** [reply]
hello  @Chicagoan2016 ,

I am used Nanoframework for Running the C# on ESP32.

1.Create A **Controller**  in  AppServerCoreHost service  
![image](https://user-images.githubusercontent.com/17121653/93429350-a56c7c80-f8de-11ea-9d27-2c5e6252b793.png)

    
*Write here any Post/Get methods in that as per your need --inside the you can used  any **CSLA BO's** 

2. You can Call Above Api from any device which support api call in my case I used **Nanoframework** because I want use C#.
refer the below link for calling api  
   .https://github.com/nanoframework/Samples/blob/master/samples/HTTP/HttpWebRequest/HttpWebRequest/Program.cs
    reference for how to call the api in nano
  *if  you can **Deserialization** the string response into DTO class which you can use for further operation inside your device.

Thanks @rockfordlhotka for considered me in this conversion.
